chalk-ir/src/family.rs

```diff
@@ -63,12 +41,17 @@ pub trait TypeFamily: Debug + Copy + Eq + Ord + Hash {
     ) -> fmt::Result;
 
     /// Create an "interned" type from `ty`. You can also use the
-    /// `Ty::intern` method, which is preferred.
-    fn intern_ty(ty: Ty<Self>) -> Self::Type;
+    /// `TyData::intern` method, which is preferred.
+    fn intern_ty(ty: TyData<Self>) -> Self::InternedType;
```

Comment:
You can't remove it from the trait -- that is how `ty.intern()` is implemented. :) I can improve the comment, though.

Reply:
Do you mean that some implementations of `TypeFamily` will end up needing it? Because Chalk compiles just fine if you remove it from the trait. None of the current `intern()` impl's use it and `Ty::new()` just calls `intern()` directly now.